server/main.lua

```diff
@@ -59,36 +59,64 @@ end)
 ----   Functions   ----
 -----------------------
 
+--- Gives the user the keys to the vehicle
+--- @param source number ID of the player
+--- @param plate string The plate number of the vehicle.
 function GiveKeys(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
     local keys = Player(source).state.keysList or {}
 
     if keys[plate] then return end
     keys[plate] = true
-    Player(source).state:set('keysList', keys, true)
-    local citizenid = getCitizenId(source)
 
-    if not citizenid then return end
+    Player(source).state:set('keysList', keys, true)
 
     if not keysList[citizenid] then
-        keysList[citizenid] = {}
+        keysList[citizenid] = {plate = true}
+    else
+        keysList[citizenid][plate] = true
     end
 
-    keysList[citizenid][plate] = true
-
     exports.qbx_core:Notify(source, locale('notify.keys_taken'))
 end
 
+RegisterNetEvent('qbx_vehiclekeys:server:giveKeys', function(source, plate)
+    if not GetInvokingResource() then return end
```

Comment:
This will restrict this event to be only handled on its side of the network. So, it should be a non-net event? Or was this intended to be a net event and this meant to be `if GetInvokingResource() then return end` to prevent it being called from the server? I'm guessing the latter since there is already a GiveKeys export. By the way, if it doesn't already exist we should introduce a client export GiveKeys to provide the API to the client. This should be an internal event only.

Reply:
Also, maybe appropriate to lib.print.warn if this condition is tripped.

Reply:
This is a relic of the qbx_adminmenu commands .
Such a call does not require strong binding of other resources to the GiveKeys function.
The problem with export is that when qbx_vehiclekeys is disabled, the qbx_core screams.
![obraz](https://github.com/Qbox-project/qbx_vehiclekeys/assets/36608116/272cc50c-75f7-4ddf-9471-e3cafe3905fc)

I have concerns about directly invoking this command from the client.

I'll remove it if it is unnecessary!

Reply:
That would be core's responsibility to check if the resource is running, no?

I agree with not exposing the export on the client, all the more reason for it to be a server side export rather than a net event.